File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/repositories/MessageRepository.java
package com.szlazakm.chatserver.repositories;

import com.szlazakm.chatserver.entities.Message;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface MessageRepository extends JpaRepository<Message, UUID> {

    List<Message> getAllByToPhoneNumber(String toPhoneNumber);
}

File: client/app/src/main/java/com/szlazakm/safechat/client/data/repositories/MessageRepository.kt
package com.szlazakm.safechat.client.data.repositories

import android.content.Context
import androidx.room.Room
import com.szlazakm.safechat.AppDatabase
import com.szlazakm.safechat.client.data.entities.MessageEntity
import com.szlazakm.safechat.client.data.entities.toTextMessage
import com.szlazakm.safechat.client.domain.Message
import javax.inject.Inject

class MessageRepository @Inject constructor(context: Context) {

    private val database = Room.databaseBuilder(
        context.applicationContext,
        AppDatabase::class.java, "app-database"
    ).fallbackToDestructiveMigration().build()

    fun getMessages(senderPhoneNumber: String, receiverPhoneNumber: String): List<Message.TextMessage> {
        return database.messageDao().getMessages(
            senderPhoneNumber,
            receiverPhoneNumber
        ).map { messageEntity -> messageEntity.toTextMessage() }
    }

    fun addMessage(message: MessageEntity) {
        database.messageDao().insertMessage(message)
    }

}